pkg/multicast/multicast_test.go

```diff
@@ -2,53 +2,190 @@ package multicast
 
 import (
 	"bytes"
+	"context"
+	"encoding/json"
+	"errors"
+	"io"
+	"io/ioutil"
+	"math"
+	"net"
+	"reflect"
+	"sort"
 	"testing"
+	"time"
 
+	"github.com/KyberNetwork/deribit-api/pkg/common"
 	"github.com/KyberNetwork/deribit-api/pkg/models"
 	"github.com/KyberNetwork/deribit-api/pkg/multicast/sbe"
 	"github.com/stretchr/testify/suite"
+	"golang.org/x/net/ipv4"
 )
 
+const (
+	instrumentsFilePath = "mock/instruments.json"
+)
+
+var errInvalidParam = errors.New("invalid params")
+
+type MockInstrumentsGetter struct{}
+
+func (m *MockInstrumentsGetter) GetInstruments(
+	ctx context.Context, params *models.GetInstrumentsParams,
+) ([]models.Instrument, error) {
+	var allIns, btcIns, ethIns []models.Instrument
+	instrumentsBytes, err := ioutil.ReadFile(instrumentsFilePath)
+	if err != nil {
+		return nil, err
+	}
+
+	err = json.Unmarshal(instrumentsBytes, &allIns)
+	if err != nil {
+		return nil, err
+	}
+	for _, ins := range allIns {
+		if ins.BaseCurrency == "BTC" {
+			btcIns = append(btcIns, ins)
+		} else if ins.BaseCurrency == "ETH" {
+			ethIns = append(ethIns, ins)
+		}
+	}
+
+	switch params.Currency {
+	case "BTC":
+		return btcIns, nil
+	case "ETH":
+		return ethIns, nil
+	default:
+		return nil, errInvalidParam
+	}
+}
+
 type MulticastTestSuite struct {
 	suite.Suite
-	m *sbe.SbeGoMarshaller
-	c *Client
+	m           *sbe.SbeGoMarshaller
+	c           *Client
+	wrongClient *Client
+	ins         []models.Instrument
+	insMap      map[uint32]models.Instrument
 }
 
 func TestMulticastTestSuite(t *testing.T) {
 	suite.Run(t, new(MulticastTestSuite))
 }
 
 func (ts *MulticastTestSuite) SetupSuite() {
+	require := ts.Require()
+	var (
+		ifname     = "not-exits-ifname"
+		ipAddrs    = []string{"239.111.111.1", "239.111.111.2", "239.111.111.3"}
+		port       = 6100
+		currencies = []string{"BTC", "ETH"}
+	)
+
 	m := sbe.NewSbeGoMarshaller()
 
-	ts.c = &Client{}
+	// Error case
+	client, err := NewClient(ifname, ipAddrs, port, &MockInstrumentsGetter{}, currencies)
+	require.Error(err)
+	require.Nil(client)
+
+	// Success case
+	client, err = NewClient("en0", ipAddrs, port, &MockInstrumentsGetter{}, currencies)
```

Comment:
Get interface index of `0` using bellow function and then get its name.

https://pkg.go.dev/net#InterfaceByIndex